core/sys/posix/fcntl.odin

```diff
@@ -0,0 +1,415 @@
+	RDWR      = log2(O_RDWR),
+	// Writing only.
+	WRONLY    = log2(O_WRONLY),
+	// Reading only.
+	// RDONLY = 0, // Default
+
+}
+O_Flags :: bit_set[O_Flag_Bits; c.int]
+
+// A mask of all the access mode bits.
+O_ACCMODE :: O_Flags{ .EXEC, .RDWR, .WRONLY }
+
+AT_Flag_Bits :: enum c.int {
+	EACCESS          = log2(AT_EACCESS),
+	SYMLINK_NOFOLLOW = log2(AT_SYMLINK_NOFOLLOW),
+	SYMLINK_FOLLOW   = log2(AT_SYMLINK_FOLLOW),
+	REMOVEDIR        = log2(AT_REMOVEDIR),
+}
+AT_Flags :: bit_set[AT_Flag_Bits; c.int]
+
+when ODIN_OS == .Darwin {
+
+	off_t  :: distinct c.int64_t
+	pid_t  :: distinct c.int32_t
+
+	F_DUPFD         :: 0
+	F_DUPFD_CLOEXEC :: 67
+	F_GETFD         :: 1
+	F_SETFD         :: 2
+	F_GETFL         :: 3
+	F_SETFL         :: 4
+	F_GETLK         :: 7
+	F_SETLK         :: 8
+	F_SETLKW        :: 9
+	F_GETOWN        :: 5
+	F_SETOWN        :: 6
+
+	FD_CLOEXEC :: 1
+
+	F_RDLCK :: 1
+	F_UNLCK :: 2
+	F_WRLCK :: 3
+
+	O_CLOEXEC   :: 0x01000000
+	O_CREAT     :: 0x00000200
+	O_DIRECTORY :: 0x00100000
+	O_EXCL      :: 0x00000800
+	O_NOCTTY    :: 0x00020000
+	O_NOFOLOW   :: 0x00000100
+	O_TRUNC     :: 0x00000400
+
+	_O_TTY_INIT :: 0
+	O_TTY_INIT  :: O_Flags{}
+
+	O_APPEND   :: 0x00000008
+	O_DSYNC    :: 0x00400000
+	O_NONBLOCK :: 0x00000004
+	O_SYNC     :: 0x0080
+
+	_O_RSYNC   :: 0
+	O_RSYNC    :: O_Flags{}
+
+	O_EXEC    :: 0x40000000
+	O_RDONLY  :: 0
+	O_RDWR    :: 0x0002
+	O_WRONLY  :: 0x0001
+
+	_O_SEARCH :: O_EXEC | O_DIRECTORY
+	O_SEARCH  :: O_Flags{ .EXEC, .DIRECTORY }
+
+	AT_FDCWD: FD: -2
+
+	AT_EACCESS          :: 0x0010
+	AT_SYMLINK_NOFOLLOW :: 0x0020
+	AT_SYMLINK_FOLLOW   :: 0x0040
+	AT_REMOVEDIR        :: 0x0080
+
+	flock :: struct {
+		l_start:  off_t,     /* [PSX] relative offset in bytes */
+		l_len:    off_t,     /* [PSX] size; if 0 then until EOF */
+		l_pid:    pid_t,     /* [PSX] process ID of the process holding the lock */
+		l_type:   Lock_Type, /* [PSX] type of lock */
+		l_whence: c.short,   /* [PSX] flag (Whence) of starting offset */
+	}
+
+} else when ODIN_OS == .FreeBSD {
+
+	off_t  :: distinct c.int64_t
+	pid_t  :: distinct c.int32_t
+
+	F_DUPFD         :: 0
+	F_DUPFD_CLOEXEC :: 17
+	F_GETFD         :: 1
+	F_SETFD         :: 2
+	F_GETFL         :: 3
+	F_SETFL         :: 4
+	F_GETLK         :: 7
+	F_SETLK         :: 8
+	F_SETLKW        :: 9
+	F_GETOWN        :: 5
+	F_SETOWN        :: 6
+
+	FD_CLOEXEC :: 1
+
+	F_RDLCK :: 1
+	F_UNLCK :: 2
+	F_WRLCK :: 3
+
+	O_CLOEXEC   :: 0x00100000
+	O_CREAT     :: 0x0200
+	O_DIRECTORY :: 0x00020000
+	O_EXCL      :: 0x0800
+	O_NOCTTY    :: 0x8000
+	O_NOFOLOW   :: 0x0100
+	O_TRUNC     :: 0x0400
+
+	_O_TTY_INIT :: 0x00080000
+	O_TTY_INIT  :: O_Flags{O_Flag_Bits(log2(_O_TTY_INIT))}
+
+	O_APPEND   :: 0x0008
+	O_DSYNC    :: 0x01000000
+	O_NONBLOCK :: 0x0004
+	O_SYNC     :: 0x0080
+	_O_RSYNC   :: 0
+	O_RSYNC    :: O_Flags{}
+
+	O_EXEC    :: 0x00040000
+	O_RDONLY  :: 0
+	O_RDWR    :: 0x0002
+	O_WRONLY  :: 0x0001
+
+	_O_SEARCH :: O_EXEC|O_DIRECTORY
+	O_SEARCH  :: O_Flags{ .EXEC, .DIRECTORY }
```

Comment:
`O_SEARCH` is the same value as `O_EXEC` on FreeBSD.

Reply:
It is technically fine like this because O_SEARCH can only be used in combination with O_DIRECTORY, supplying O_SEARCH without O_DIRECTORY would return EINVAL afaik.

But, I see no reason to not make it the exact same so I will make that change.